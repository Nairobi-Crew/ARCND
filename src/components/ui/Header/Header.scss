@import "src/common/vars";

.header {
  padding: 15px 130px;

  &__title {
    color: $fc;
    font: {
      family: $ff;
      weight: 600;
      size: $f-xxl;
    }
  }
}
