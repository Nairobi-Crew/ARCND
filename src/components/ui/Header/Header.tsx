import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';

const Header = () => (
  <header className="header">
    <Link to="/" className="header__title">
      <h1>ARCND</h1>
    </Link>
  </header>
);

export default Header;
