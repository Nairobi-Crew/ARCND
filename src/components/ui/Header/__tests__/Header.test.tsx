/* eslint-disable no-undef */
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import React from 'react';
import Header from 'UI/Header/Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Render Header', () => {
  const Component = () => (<Router><Header /></Router>);
  it('Render correct', () => {
    const inputComponent = renderer.create(<Component />).toJSON();
    expect(inputComponent).toMatchSnapshot();
  });

  it('test className', () => {
    const wrapper = mount(<Component />);
    expect(wrapper.find('.header')).toHaveLength(1);
    expect(wrapper.find('.header').find('.header__title')).toHaveLength(3);
  });
});
