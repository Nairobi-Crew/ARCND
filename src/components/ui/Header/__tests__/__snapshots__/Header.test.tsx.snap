// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Render correct 1`] = `
<header
  className="header"
>
  <a
    className="header__title"
    href="/"
    onClick={[Function]}
  >
    <h1>
      ARCND
    </h1>
  </a>
</header>
`;
